pipeline/metadata/flatten_satellite.py

```diff
@@ -565,3 +502,63 @@ def _process_satellite_v2_control(
         row['received'] = all_received
 
         yield row
+
+
+class FlattenBlockpages(beam.DoFn):
+  """DoFn class for flattening lines of blockpage text into Rows."""
+
+  def setup(self) -> None:
+    #pylint: disable=attribute-defined-outside-init
+    self.blockpage_matcher = BlockpageMatcher()
+    #pylint: enable=attribute-defined-outside-init
+
+  def process(self, element: Tuple[str, str]) -> Iterator[BlockpageRow]:
+    (filename, line) = element
+
+    try:
+      scan = json.loads(line)
+    except json.decoder.JSONDecodeError as e:
+      logging.warning('JSONDecodeError: %s\nFilename: %s\n%s\n', e, filename,
+                      line)
+      return
+
+    yield from self._process_satellite_blockpages(scan, filename)
+
+  def _process_satellite_blockpages(self, blockpage_entry: ResponsesEntry,
```

Comment:
It's called that per your request in [PR116](https://github.com/censoredplanet/censoredplanet-analysis/pull/116) [commit](https://github.com/censoredplanet/censoredplanet-analysis/commit/cdfd926f763c4784d1a44b42206ea46d2ca70081) because the response and blockpage data was processed in the same path.

I've split out a `BlockpageEntry` type to match `ResponsesEntry`.